sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/ReactorSenderTest.java

```diff
@@ -117,6 +122,30 @@ public void testLinkSize() throws IOException {
         verify(sender, times(1)).getRemoteMaxMessageSize();
     }
 
+    /**
+     * Testing that we can send message with transaction.
+     */
+    @Test
+    public void testSendWithTransaction() {
+        Message message = Proton.message();
+        message.setMessageId("id");
+        message.setBody(new AmqpValue("hello"));
+
+        AmqpTransaction transaction = new AmqpTransaction(ByteBuffer.wrap("1".getBytes()));
+
+        ReactorSender reactorSender = new ReactorSender(entityPath, sender, handler, reactorProvider, tokenManager,
```

Comment:
I'm confused what you are actually testing here... Just that it eventually calls the send() overload?

Reply:
We're missing tests for using the delivery state on success and not.

Reply:
Or that the transactionDeliveryState is set on a delivery.